File: pdu/python/hako_srv_msgs/pdu_cdr_conv_AddTwoIntsRequest.py
from .pdu_pytype_AddTwoIntsRequest import AddTwoIntsRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_AddTwoIntsRequest(writer: CdrWriter, src: AddTwoIntsRequest):
    writer.write_int64(src.a)
    writer.write_int64(src.b)


def cdr_body_to_py_AddTwoIntsRequest(reader: CdrReader, dst: AddTwoIntsRequest):
    dst.a = reader.read_int64()
    dst.b = reader.read_int64()
    return dst


def py_to_cdr_AddTwoIntsRequest(src: AddTwoIntsRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_AddTwoIntsRequest(writer, src)
    return writer.bytes()


def cdr_to_py_AddTwoIntsRequest(cdr_payload) -> AddTwoIntsRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = AddTwoIntsRequest()
    return cdr_body_to_py_AddTwoIntsRequest(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_AddTwoIntsRequest.js
import { AddTwoIntsRequest } from './pdu_jstype_AddTwoIntsRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduAddTwoIntsRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { AddTwoIntsRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_int64(src.a);
        writer.write_int64(src.b);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { AddTwoIntsRequest } dst
     * @returns { AddTwoIntsRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.a = reader.read_int64();
        dst.b = reader.read_int64();
        return dst;
    }

    /**
     * @param { AddTwoIntsRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { AddTwoIntsRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new AddTwoIntsRequest());
    }
}
